src/Command/Push/PushArtifactCommand.php

```diff
@@ -441,10 +450,31 @@ protected function determineDestinationGitBranch() {
     if ($env_var = getenv('ACLI_PUSH_ARTIFACT_DESTINATION_GIT_BRANCH')) {
       return $env_var;
     }
+
+    $this->environment = $this->determineEnvironment($this->input, $this->output, FALSE);
     if (strpos($this->environment->vcs->path, 'tags') === 0) {
       throw new AcquiaCliException("You cannot push to an environment that has a git tag deployed to it. Environment {$this->environment->name} has {$this->environment->vcs->path} deployed. Please select a different environment.");
```

Comment:
Is this error message unnecessary now?

Reply:
I think it's still necessary. The naming of variables is tricky and maybe a bit misleading. The "destination-git-branch" which is inferred from the environment is actually used as the source branch and the destination branch when a tag is not being used. 

When a tag is being used, destination-git-branch is used as the source branch and the tag is used as the destination git ref.

Reply:
* `acli push:artifact`
    * prompts to choose environment. the branch of the chosen environment will be the `--destination-git-branch`
    * `--destination-git-branch` is used as the source and destination branch
* `acli push:artifact --destination-git-branch=[branch]`
    * `--destination-git-branch` is used as the source and destination branch
* `acli push:artifact --destination-git-branch=[branch] --tag-name=[tag]`
    * `--destination-git-branch` is used as the source and destination branch

Reply:
I've updated this.